pkg/queue/health/probe_test.go

```diff
@@ -19,25 +19,172 @@ package health
 import (
 	"net/http"
 	"net/http/httptest"
+	"net/url"
 	"testing"
 	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/knative/serving/pkg/network"
+	"github.com/knative/serving/pkg/queue"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
 )
 
 func TestTCPProbe(t *testing.T) {
 	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(http.StatusOK)
 	}))
 	defer server.Close()
-	serverAddr := server.Listener.Addr().String()
+	config := TCPProbeConfigOptions{
+		Address:       server.Listener.Addr().String(),
+		SocketTimeout: time.Second,
+	}
+	// Connecting to the server should work
+	if err := TCPProbe(config); err != nil {
+		t.Errorf("Expected probe to succeed but it failed with %v", err)
+	}
 
+	// Close the server so probing fails afterwards
+	server.Close()
+	if err := TCPProbe(config); err == nil {
+		t.Error("Expected probe to fail but it didn't")
+	}
+}
+
+func TestHTTPProbeSuccess(t *testing.T) {
+	var gotHeader, kubeletHeader corev1.HTTPHeader
+	expectedHeader := corev1.HTTPHeader{
+		Name:  "Testkey",
+		Value: "Testval",
+	}
+	var gotPath string
+	expectedPath := "/health"
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		for headerKey, headerValue := range r.Header {
+			// Filtering for expectedHeader.TestKey to avoid other HTTP probe headers
+			if expectedHeader.Name == headerKey {
+				gotHeader = corev1.HTTPHeader{Name: headerKey, Value: headerValue[0]}
+			}
+
+			if headerKey == network.KubeletProbeHeaderName {
```

Comment:
This is the header we set to tell ourselves it's a kubelet probe we configured (workaround for Istio mTLS which has the bug I'm trying to avoid).

I think we want to check `network.IsKubeletProbe(r)` before returning `StatusOK` below.  Should fail as-is, but pass when we get it right.